### ISSUE TITLE
travis build with python binding

### ISSUE THREAD

**Issue by vhb0fldb** [dreal/dreal4#204]
This is my .travis.yml
```
before_install:
  - sudo apt-get update
  - sudo apt-get install apt software-properties-common python3-pip -y
  - sudo add-apt-repository ppa:dreal/dreal -y
  - sudo apt update
  - wget https://dl.bintray.com/dreal/dreal/dreal_4.20.04.1_amd64.deb
  - sudo dpkg -i dreal_4.20.04.1_amd64.deb || true
  - sudo apt install -f -y
install:
  - pip3 install dreal
```

I tried too to use the instructions on your README:
```
sudo apt-get update
sudo apt-get install curl -y
curl -fsSL https://raw.githubusercontent.com/dreal/dreal4/master/setup/ubuntu/18.04/install.sh | sudo bash
```
The error is always
```
Hint: make sure your test modules/packages have valid Python names.

Traceback:


    from dreal import *

../../../virtualenv/python3.6.7/lib/python3.6/site-packages/dreal/__init__.py:7: in <module>

    from dreal._dreal_py import *

E   ImportError: libClpSolver.so.1: cannot open shared object file: No such file or directory

!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
```
The project uses dreal4 in python3. How do I install dreal4 correctly with its python bindings for travis? The travis script worked until I added dreal4

**Comment by soonho-tri** [comment]
Hi, can you point me the travis URL if you can?

**Comment by vhb0fldb** [comment]
The repo is private; I can post more details from it.

```
0.00s$ source ~/virtualenv/python3.6/bin/activate

$ python --version

Python 3.6.7

$ pip --version

pip 19.0.3 from /home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/pip (python 3.6)
before_install.1

1.95s$ sudo apt-get update
before_install.2

7.17s$ sudo apt-get install apt software-properties-common python3-pip -y
before_install.3

1.61s$ sudo add-apt-repository ppa:dreal/dreal -y
before_install.4

5.24s$ sudo apt update
before_install.5

0.51s$ wget https://dl.bintray.com/dreal/dreal/dreal_4.20.04.1_amd64.deb
before_install.6

0.15s$ sudo dpkg -i dreal_4.20.04.1_amd64.deb || true
before_install.7

15.83s$ sudo apt install -f -y
install

112.34s$ pip3 install -r requirements.txt

5.90s$ pytest

============================= test session starts ==============================

platform linux -- Python 3.6.7, pytest-4.3.1, py-1.7.0, pluggy-0.8.0

rootdir: /home/travis/build/, inifile:

collected 7 items / 1 errors / 6 selected                                      

==================================== ERRORS ====================================



Hint: make sure your test modules/packages have valid Python names.

Traceback:


    from dreal import *

../../../virtualenv/python3.6.7/lib/python3.6/site-packages/dreal/__init__.py:7: in <module>

    from dreal._dreal_py import *

E   ImportError: libClpSolver.so.1: cannot open shared object file: No such file or directory

!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
```

I can attach the full log if it's useful

**Comment by soonho-tri** [comment]
Was there any error when you execute the line?

```sh
curl -fsSL https://raw.githubusercontent.com/dreal/dreal4/master/setup/ubuntu/18.04/install.sh | sudo bash
```

Also, please check if your system has `coinor-libclp-dev` after running the script. That's the one providing `libClpSolver.so.1`.

**Comment by vhb0fldb** [comment]
It seems it is being installed. 

```
0.15s$ sudo dpkg -i dreal_4.20.04.1_amd64.deb || true

Selecting previously unselected package dreal.

(Reading database ... 124285 files and directories currently installed.)

Preparing to unpack dreal_4.20.04.1_amd64.deb ...

Unpacking dreal (4.20.04.1) ...

dpkg: dependency problems prevent configuration of dreal:

 dreal depends on coinor-libclp-dev; however:

  Package coinor-libclp-dev is not installed.

 dreal depends on libibex-dev; however:

  Package libibex-dev is not installed.

 dreal depends on libnlopt-dev; however:

  Package libnlopt-dev is not installed.

dpkg: error processing package dreal (--install):

 dependency problems - leaving unconfigured

Errors were encountered while processing:

 dreal
before_install.7

15.83s$ sudo apt install -f -y

Reading package lists... Done

Building dependency tree       

Reading state information... Done

Correcting dependencies... Done

The following additional packages will be installed:

  coinor-libclp-dev coinor-libclp1 coinor-libcoinutils-dev

  coinor-libcoinutils3v5 coinor-libosi-dev coinor-libosi1v5 cpp-5 g++-5 gcc-5

  gcc-5-base gfortran gfortran-5 libasan2 libatomic1 libblas-common

  libblas-dev libblas3 libcc1-0 libcilkrts5 libgcc-5-dev libgfortran-5-dev

  libgfortran3 libgomp1 libibex-dev libitm1 liblapack-dev liblapack3 liblsan0

  libmpx0 libnlopt-dev libnlopt0 libquadmath0 libstdc++-5-dev libstdc++6

  libtsan0 libubsan0

Suggested packages:

  gcc-5-locales g++-5-multilib gcc-5-doc libstdc++6-5-dbg gcc-5-multilib

  libgcc1-dbg libgomp1-dbg libitm1-dbg libatomic1-dbg libasan2-dbg

  liblsan0-dbg libtsan0-dbg libubsan0-dbg libcilkrts5-dbg libmpx0-dbg

  libquadmath0-dbg gfortran-multilib gfortran-doc gfortran-5-multilib

  gfortran-5-doc libgfortran3-dbg liblapack-doc-man liblapack-doc

  libstdc++-5-doc

The following NEW packages will be installed:

  coinor-libclp-dev coinor-libclp1 coinor-libcoinutils-dev

  coinor-libcoinutils3v5 coinor-libosi-dev coinor-libosi1v5 gfortran

  gfortran-5 libblas-common libblas-dev libblas3 libgfortran-5-dev

  libgfortran3 libibex-dev liblapack-dev liblapack3 libnlopt-dev libnlopt0

The following packages will be upgraded:

  cpp-5 g++-5 gcc-5 gcc-5-base libasan2 libatomic1 libcc1-0 libcilkrts5

  libgcc-5-dev libgomp1 libitm1 liblsan0 libmpx0 libquadmath0 libstdc++-5-dev

  libstdc++6 libtsan0 libubsan0

18 upgraded, 18 newly installed, 0 to remove and 263 not upgraded.

1 not fully installed or removed.

Need to get 47.9 MB of archives.

After this operation, 69.7 MB of additional disk space will be used.

Get:1 http://archive.ubuntu.com/ubuntu xenial/main amd64 libblas-common amd64 3.6.0-2ubuntu2 [5,342 B]

Get:2 http://ppa.launchpad.net/dreal/dreal/ubuntu xenial/main amd64 libibex-dev amd64 2.7.4.20200415183212.git0112b423c94515302c551c478c30e98ff442752d~16.04 [1,295 kB]

Get:3 http://archive.ubuntu.com/ubuntu xenial/main amd64 libblas3 amd64 3.6.0-2ubuntu2 [147 kB]

Get:4 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libgomp1 amd64 5.4.0-6ubuntu1~16.04.12 [55.2 kB]

Get:5 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libitm1 amd64 5.4.0-6ubuntu1~16.04.12 [27.4 kB]

Get:6 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libatomic1 amd64 5.4.0-6ubuntu1~16.04.12 [8,892 B]

Get:7 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libasan2 amd64 5.4.0-6ubuntu1~16.04.12 [265 kB]

Get:8 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 liblsan0 amd64 5.4.0-6ubuntu1~16.04.12 [105 kB]

Get:9 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libtsan0 amd64 5.4.0-6ubuntu1~16.04.12 [244 kB]

Get:10 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libubsan0 amd64 5.4.0-6ubuntu1~16.04.12 [95.3 kB]

Get:11 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libcilkrts5 amd64 5.4.0-6ubuntu1~16.04.12 [40.0 kB]

Get:12 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libmpx0 amd64 5.4.0-6ubuntu1~16.04.12 [9,762 B]

Get:13 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libquadmath0 amd64 5.4.0-6ubuntu1~16.04.12 [131 kB]

Get:14 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 cpp-5 amd64 5.4.0-6ubuntu1~16.04.12 [7,783 kB]

Get:15 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libcc1-0 amd64 5.4.0-6ubuntu1~16.04.12 [38.8 kB]

Get:16 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 g++-5 amd64 5.4.0-6ubuntu1~16.04.12 [8,430 kB]

Get:17 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 gcc-5 amd64 5.4.0-6ubuntu1~16.04.12 [8,612 kB]

Get:18 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libgcc-5-dev amd64 5.4.0-6ubuntu1~16.04.12 [2,239 kB]

Get:19 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libstdc++-5-dev amd64 5.4.0-6ubuntu1~16.04.12 [1,428 kB]

Get:20 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 gcc-5-base amd64 5.4.0-6ubuntu1~16.04.12 [17.4 kB]

Get:21 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libstdc++6 amd64 5.4.0-6ubuntu1~16.04.12 [393 kB]

Get:22 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libgfortran3 amd64 5.4.0-6ubuntu1~16.04.12 [260 kB]

Get:23 http://archive.ubuntu.com/ubuntu xenial/main amd64 liblapack3 amd64 3.6.0-2ubuntu2 [1,938 kB]

Get:24 http://archive.ubuntu.com/ubuntu xenial/universe amd64 coinor-libcoinutils3v5 amd64 2.9.15-3.1 [465 kB]

Get:25 http://archive.ubuntu.com/ubuntu xenial/universe amd64 coinor-libosi1v5 amd64 0.106.9-2 [280 kB]

Get:26 http://archive.ubuntu.com/ubuntu xenial/universe amd64 coinor-libclp1 amd64 1.15.10-1build2 [730 kB]

Get:27 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libgfortran-5-dev amd64 5.4.0-6ubuntu1~16.04.12 [291 kB]

Get:28 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 gfortran-5 amd64 5.4.0-6ubuntu1~16.04.12 [8,158 kB]

Get:29 http://archive.ubuntu.com/ubuntu xenial/main amd64 gfortran amd64 4:5.3.1-1ubuntu1 [1,288 B]

Get:30 http://archive.ubuntu.com/ubuntu xenial/main amd64 libblas-dev amd64 3.6.0-2ubuntu2 [153 kB]

Get:31 http://archive.ubuntu.com/ubuntu xenial/main amd64 liblapack-dev amd64 3.6.0-2ubuntu2 [1,948 kB]

Get:32 http://archive.ubuntu.com/ubuntu xenial/universe amd64 coinor-libcoinutils-dev amd64 2.9.15-3.1 [781 kB]

Get:33 http://archive.ubuntu.com/ubuntu xenial/universe amd64 coinor-libosi-dev amd64 0.106.9-2 [337 kB]

Get:34 http://archive.ubuntu.com/ubuntu xenial/universe amd64 coinor-libclp-dev amd64 1.15.10-1build2 [887 kB]

Get:35 http://archive.ubuntu.com/ubuntu xenial/universe amd64 libnlopt0 amd64 2.4.2+dfsg-2 [165 kB]

Get:36 http://archive.ubuntu.com/ubuntu xenial/universe amd64 libnlopt-dev amd64 2.4.2+dfsg-2 [174 kB]

Fetched 47.9 MB in 5s (8,400 kB/s)

Extracting templates from packages: 100%

7mon.

(Reading database ... 124331 files and directories currently installed.)

Preparing to unpack .../libblas-common_3.6.0-2ubuntu2_amd64.deb ...

7Progress: [  0%] [..........................................................] 8Unpacking libblas-common (3.6.0-2ubuntu2) ...

7Progress: [  1%] [#.........................................................] 8Selecting previously unselected package libblas3.

Preparing to unpack .../libblas3_3.6.0-2ubuntu2_amd64.deb ...

7Progress: [  2%] [##........................................................] 8Unpacking libblas3 (3.6.0-2ubuntu2) ...

7Progress: [  3%] [##........................................................] 8Preparing to unpack .../libgomp1_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

Unpacking libgomp1:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [  4%] [###.......................................................] 8Preparing to unpack .../libitm1_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [  5%] [####......................................................] 8Unpacking libitm1:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [  6%] [####......................................................] 8Preparing to unpack .../libatomic1_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [  7%] [#####.....................................................] 8Unpacking libatomic1:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [  8%] [#####.....................................................] 8Preparing to unpack .../libasan2_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

Unpacking libasan2:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [  9%] [######....................................................] 8Preparing to unpack .../liblsan0_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 10%] [######....................................................] 8Unpacking liblsan0:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 11%] [#######...................................................] 8Preparing to unpack .../libtsan0_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

Unpacking libtsan0:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 12%] [########..................................................] 87Progress: [ 13%] [########..................................................] 8Preparing to unpack .../libubsan0_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

Unpacking libubsan0:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 14%] [#########.................................................] 8Preparing to unpack .../libcilkrts5_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 15%] [#########.................................................] 8Unpacking libcilkrts5:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 16%] [##########................................................] 8Preparing to unpack .../libmpx0_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

Unpacking libmpx0:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 17%] [###########...............................................] 8Preparing to unpack .../libquadmath0_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 18%] [###########...............................................] 8Unpacking libquadmath0:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 19%] [############..............................................] 8Preparing to unpack .../cpp-5_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 20%] [############..............................................] 8Unpacking cpp-5 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 21%] [#############.............................................] 8Preparing to unpack .../libcc1-0_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

Unpacking libcc1-0:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 22%] [#############.............................................] 8Preparing to unpack .../g++-5_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 23%] [##############............................................] 8Unpacking g++-5 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 24%] [###############...........................................] 8Preparing to unpack .../gcc-5_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 25%] [###############...........................................] 8Unpacking gcc-5 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 26%] [################..........................................] 8Preparing to unpack .../libgcc-5-dev_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

Unpacking libgcc-5-dev:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 27%] [################..........................................] 8Preparing to unpack .../libstdc++-5-dev_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 28%] [#################.........................................] 8Unpacking libstdc++-5-dev:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 29%] [##################........................................] 8Preparing to unpack .../gcc-5-base_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

Unpacking gcc-5-base:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 30%] [##################........................................] 8Processing triggers for libc-bin (2.23-0ubuntu11) ...

Processing triggers for man-db (2.7.5-1) ...

Processing triggers for ccache (3.2.4-1) ...

Updating symlinks in /usr/lib/ccache ...

Setting up gcc-5-base:amd64 (5.4.0-6ubuntu1~16.04.12) ...

(Reading database ... 124340 files and directories currently installed.)

Preparing to unpack .../libstdc++6_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

Unpacking libstdc++6:amd64 (5.4.0-6ubuntu1~16.04.12) over (5.4.0-6ubuntu1~16.04.11) ...

7Progress: [ 33%] [####################......................................] 8Processing triggers for libc-bin (2.23-0ubuntu11) ...

Setting up libstdc++6:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 34%] [####################......................................] 8Processing triggers for libc-bin (2.23-0ubuntu11) ...

Selecting previously unselected package libgfortran3:amd64.

(Reading database ... 124340 files and directories currently installed.)

Preparing to unpack .../libgfortran3_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 35%] [#####################.....................................] 8Unpacking libgfortran3:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 36%] [######################....................................] 8Selecting previously unselected package liblapack3.

Preparing to unpack .../liblapack3_3.6.0-2ubuntu2_amd64.deb ...

Unpacking liblapack3 (3.6.0-2ubuntu2) ...

7Progress: [ 37%] [######################....................................] 87Progress: [ 38%] [#######################...................................] 8Selecting previously unselected package coinor-libcoinutils3v5.

Preparing to unpack .../coinor-libcoinutils3v5_2.9.15-3.1_amd64.deb ...

Unpacking coinor-libcoinutils3v5 (2.9.15-3.1) ...

7Progress: [ 39%] [#######################...................................] 8Selecting previously unselected package coinor-libosi1v5.

Preparing to unpack .../coinor-libosi1v5_0.106.9-2_amd64.deb ...

7Progress: [ 40%] [########################..................................] 8Unpacking coinor-libosi1v5 (0.106.9-2) ...

7Progress: [ 41%] [########################..................................] 8Selecting previously unselected package coinor-libclp1.

Preparing to unpack .../coinor-libclp1_1.15.10-1build2_amd64.deb ...

Unpacking coinor-libclp1 (1.15.10-1build2) ...

7Progress: [ 42%] [#########################.................................] 8Selecting previously unselected package libgfortran-5-dev:amd64.

Preparing to unpack .../libgfortran-5-dev_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 43%] [##########################................................] 8Unpacking libgfortran-5-dev:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 44%] [##########################................................] 8Selecting previously unselected package gfortran-5.

Preparing to unpack .../gfortran-5_5.4.0-6ubuntu1~16.04.12_amd64.deb ...

7Progress: [ 45%] [###########################...............................] 8Unpacking gfortran-5 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 46%] [###########################...............................] 8Selecting previously unselected package gfortran.

Preparing to unpack .../gfortran_4%3a5.3.1-1ubuntu1_amd64.deb ...

Unpacking gfortran (4:5.3.1-1ubuntu1) ...

7Progress: [ 47%] [############################..............................] 8Selecting previously unselected package libblas-dev.

Preparing to unpack .../libblas-dev_3.6.0-2ubuntu2_amd64.deb ...

7Progress: [ 48%] [#############################.............................] 8Unpacking libblas-dev (3.6.0-2ubuntu2) ...

7Progress: [ 49%] [#############################.............................] 8Selecting previously unselected package liblapack-dev.

Preparing to unpack .../liblapack-dev_3.6.0-2ubuntu2_amd64.deb ...

7Progress: [ 50%] [#############################.............................] 8Unpacking liblapack-dev (3.6.0-2ubuntu2) ...

7Progress: [ 51%] [##############################............................] 8Selecting previously unselected package coinor-libcoinutils-dev.

Preparing to unpack .../coinor-libcoinutils-dev_2.9.15-3.1_amd64.deb ...

Unpacking coinor-libcoinutils-dev (2.9.15-3.1) ...

7Progress: [ 52%] [###############################...........................] 8Selecting previously unselected package coinor-libosi-dev.

Preparing to unpack .../coinor-libosi-dev_0.106.9-2_amd64.deb ...

7Progress: [ 53%] [###############################...........................] 8Unpacking coinor-libosi-dev (0.106.9-2) ...

7Progress: [ 54%] [################################..........................] 8Selecting previously unselected package coinor-libclp-dev.

Preparing to unpack .../coinor-libclp-dev_1.15.10-1build2_amd64.deb ...

Unpacking coinor-libclp-dev (1.15.10-1build2) ...

7Progress: [ 55%] [#################################.........................] 8Selecting previously unselected package libibex-dev.

Preparing to unpack .../libibex-dev_2.7.4.20200415183212.git0112b423c94515302c551c478c30e98ff442752d~16.04_amd64.deb ...

7Progress: [ 56%] [#################################.........................] 8Unpacking libibex-dev (2.7.4.20200415183212.git0112b423c94515302c551c478c30e98ff442752d~16.04) ...

7Progress: [ 57%] [##################################........................] 8Selecting previously unselected package libnlopt0:amd64.

Preparing to unpack .../libnlopt0_2.4.2+dfsg-2_amd64.deb ...

7Progress: [ 58%] [##################################........................] 8Unpacking libnlopt0:amd64 (2.4.2+dfsg-2) ...

7Progress: [ 59%] [###################################.......................] 8Selecting previously unselected package libnlopt-dev.

Preparing to unpack .../libnlopt-dev_2.4.2+dfsg-2_amd64.deb ...

Unpacking libnlopt-dev (2.4.2+dfsg-2) ...

7Progress: [ 60%] [###################################.......................] 8Processing triggers for libc-bin (2.23-0ubuntu11) ...

Processing triggers for ccache (3.2.4-1) ...

Updating symlinks in /usr/lib/ccache ...

Processing triggers for man-db (2.7.5-1) ...

Setting up libblas-common (3.6.0-2ubuntu2) ...

7Progress: [ 61%] [####################################......................] 8Setting up libblas3 (3.6.0-2ubuntu2) ...

7Progress: [ 62%] [#####################################.....................] 8update-alternatives: using /usr/lib/libblas/libblas.so.3 to provide /usr/lib/libblas.so.3 (libblas.so.3) in auto mode

7Progress: [ 63%] [#####################################.....................] 8Setting up libgomp1:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 64%] [######################################....................] 8Setting up libitm1:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 65%] [######################################....................] 8Setting up libatomic1:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 66%] [#######################################...................] 8Setting up libasan2:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 67%] [########################################..................] 8Setting up liblsan0:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 68%] [########################################..................] 8Setting up libtsan0:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 69%] [#########################################.................] 8Setting up libubsan0:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 70%] [#########################################.................] 8Setting up libcilkrts5:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 71%] [##########################################................] 8Setting up libmpx0:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 72%] [##########################################................] 8Setting up libquadmath0:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 73%] [###########################################...............] 8Setting up cpp-5 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 74%] [############################################..............] 87Progress: [ 75%] [############################################..............] 8Setting up libcc1-0:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 76%] [#############################################.............] 8Setting up libgcc-5-dev:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 77%] [#############################################.............] 8Setting up gcc-5 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 78%] [##############################################............] 8Setting up libstdc++-5-dev:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 79%] [###############################################...........] 8Setting up g++-5 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 80%] [###############################################...........] 8Setting up libgfortran3:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 81%] [################################################..........] 8Setting up liblapack3 (3.6.0-2ubuntu2) ...

7Progress: [ 82%] [################################################..........] 8update-alternatives: using /usr/lib/lapack/liblapack.so.3 to provide /usr/lib/liblapack.so.3 (liblapack.so.3) in auto mode

Setting up coinor-libcoinutils3v5 (2.9.15-3.1) ...

7Progress: [ 83%] [#################################################.........] 8Setting up coinor-libosi1v5 (0.106.9-2) ...

7Progress: [ 84%] [#################################################.........] 8Setting up coinor-libclp1 (1.15.10-1build2) ...

7Progress: [ 85%] [##################################################........] 8Setting up libgfortran-5-dev:amd64 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 86%] [###################################################.......] 8Setting up gfortran-5 (5.4.0-6ubuntu1~16.04.12) ...

7Progress: [ 87%] [###################################################.......] 87Progress: [ 88%] [####################################################......] 8Setting up gfortran (4:5.3.1-1ubuntu1) ...

update-alternatives: using /usr/bin/gfortran to provide /usr/bin/f95 (f95) in auto mode

update-alternatives: using /usr/bin/gfortran to provide /usr/bin/f77 (f77) in auto mode

7Progress: [ 89%] [####################################################......] 8Setting up libblas-dev (3.6.0-2ubuntu2) ...

update-alternatives: using /usr/lib/libblas/libblas.so to provide /usr/lib/libblas.so (libblas.so) in auto mode

7Progress: [ 90%] [#####################################################.....] 8Setting up liblapack-dev (3.6.0-2ubuntu2) ...

update-alternatives: using /usr/lib/lapack/liblapack.so to provide /usr/lib/liblapack.so (liblapack.so) in auto mode

7Progress: [ 91%] [#####################################################.....] 8Setting up coinor-libcoinutils-dev (2.9.15-3.1) ...

7Progress: [ 92%] [######################################################....] 8Setting up coinor-libosi-dev (0.106.9-2) ...

7Progress: [ 93%] [#######################################################...] 8Setting up coinor-libclp-dev (1.15.10-1build2) ...

7Progress: [ 94%] [#######################################################...] 8Setting up libibex-dev (2.7.4.20200415183212.git0112b423c94515302c551c478c30e98ff442752d~16.04) ...

7Progress: [ 95%] [########################################################..] 8Setting up libnlopt0:amd64 (2.4.2+dfsg-2) ...

7Progress: [ 96%] [########################################################..] 8Setting up libnlopt-dev (2.4.2+dfsg-2) ...

7Progress: [ 97%] [#########################################################.] 8Setting up dreal (4.20.04.1) ...

Processing triggers for libc-bin (2.23-0ubuntu11) ...

7
install

112.34s$ pip3 install -r requirements.txt

5.90s$ pytest
```

**Comment by soonho-tri** [comment]
From the last log, everything looks fine.

If you set up a public repo + travis and reproduce the same problem, I can take a closer look.

**Comment by vhb0fldb** [comment]
Do you have any reference repository using python3 I can take a look at?

**Comment by soonho-tri** [comment]
There is https://github.com/dreal/homebrew-dreal/runs/614611913?check_suite_focus=true . But it runs on macOS.

**Comment by soonho-tri** [comment]
I guess you can first check if you can reproduce it locally.

**Comment by vhb0fldb** [comment]
Reproduced at https://travis-ci.com/github/vhb0fldb/dreal from https://github.com/vhb0fldb/dreal

**Comment by soonho-tri** [comment]
Thanks.  I see the problem. In the meantime, you can use https://github.com/vhb0fldb/dreal/pull/1 as a workaround. I'll release a new wheel and ping you here.

**Comment by vhb0fldb** [comment]
Your change fixed the issue thanks!

**Comment by soonho-tri** [comment]
Good.

Python-3.6 bindings are built in a machine running Ubuntu-18.04 (because Xenial's default python version is python-3.5), and it's not backward compatible in Ubuntu-16.04 for now. In the next version, it will be fixed.

**Comment by vhb0fldb** [comment]
Thank you for your support. I will close this issue.

**Comment by soonho-tri** [comment]
FYI, I've just released `4.20.4.2` in PyPI. It should be working in Ubuntu-16.04 (Xenial).

https://pypi.org/project/dreal/4.20.4.2
